tests/hbacrule/test_hbacrule_categories.yml

```diff
@@ -109,6 +109,66 @@
     register: result
     failed_when: result.changed
 
+  - name: Ensure `user` cannot be added if usercategory is `all`.
+    ipahbacrule:
+      ipaadmin_password: SomeADMINpassword
+      name: allusers
+      user: shouldfail01
+      usercategory: "all"
+    register: result
+    ignore_errors: yes
+    failed_when: not result.failed
```

Comment:
Extend this to something like

    failed_when: not result.failed or
                 "cannot be added when user category='all'" not in result.msg

Same for the following with `host` and `service` in `result.msg`. Also for `tests/hbacrule/test_hbacrule_categories.yml`.

Reply:
Fixed.